stripe/src/main/java/com/stripe/android/model/PaymentIntent.kt

```diff
@@ -252,6 +253,22 @@ data class PaymentIntent internal constructor(
         }
     }
 
+    internal data class ClientSecret(internal val value: String) {
+        internal val paymentIntentId: String =
+            value.split("_secret".toRegex())
+                .dropLastWhile { it.isEmpty() }.toTypedArray()[0]
```

Comment:
not entirely sure how this works but wondering if there is a case where we end up with an empty array here? Or do we ensure an empty array will never be the case with the `isEmpty()`

Reply:
I'm not sure if there are any edge cases this isn't handling, but given that it's the existing logic for obtaining an ID from a client secret, I'm going to leave it as is.